bot/exts/backend/branding/_cog.py

```diff
@@ -1,566 +1,654 @@
+
+        We cache `event` information to ensure that we:
+        * Remember which event we're currently in across restarts
+        * Provide an on-demand information embed without re-querying the branding repository
 
-        If the current season is not the evergreen, and lacks at least one asset,
-        we use the evergreen seasonal dir as fallback for missing assets.
+        An event change should always be handled via this function, as it ensures that the cache is populated.
 
-        Finally, if neither the seasonal nor fallback branding directories contain
-        an asset, it will simply be ignored.
+        The #changelog notification is sent only if `event` differs from the currently cached event.
 
-        Return True if the branding has changed. This will be the case when we enter
-        a new season, or when something changes in the current seasons's directory
-        in the branding repository.
+        Returns a 2-tuple indicating whether the banner, and the icon, were applied successfully.
         """
-        old_branding = (self.banner, self.available_icons)
-        seasonal_dir = await self._get_files(self.current_season.branding_path, include_dirs=True)
+        log.debug(f"Entering event: {event.path}")
 
-        # Only make a call to the fallback directory if there is something to be gained
-        branding_incomplete = any(
-            asset not in seasonal_dir
-            for asset in (_constants.FILE_BANNER, _constants.FILE_AVATAR, _constants.SERVER_ICONS)
-        )
-        if branding_incomplete and self.current_season is not _seasons.SeasonBase:
-            fallback_dir = await self._get_files(
-                _seasons.SeasonBase.branding_path, include_dirs=True
-            )
-        else:
-            fallback_dir = {}
+        banner_success = await self.apply_banner(event.banner)  # Only one asset ~ apply directly
 
-        # Resolve assets in this directory, None is a safe value
-        self.banner = (
-            seasonal_dir.get(_constants.FILE_BANNER)
-            or fallback_dir.get(_constants.FILE_BANNER)
-        )
+        await self.initiate_icon_rotation(event.icons)  # Prepare a new rotation
+        icon_success = await self.rotate_icons()  # Apply an icon from the new rotation
+
+        # This will only be False in the case of a manual same-event re-synchronisation
+        event_changed = event.path != await self.cache_information.get("event_path")
 
-        # Now resolve server icons by making a call to the proper sub-directory
-        if _constants.SERVER_ICONS in seasonal_dir:
-            icons_dir = await self._get_files(
-                f"{self.current_season.branding_path}/{_constants.SERVER_ICONS}"
-            )
-            self.available_icons = list(icons_dir.values())
+        # Cache event identity to avoid re-entry in case of restart
+        await self.cache_information.set("event_path", event.path)
```

Comment:
That's not entirely correct, the event description can change even if the event itself doesn't. That's why it's written to the cache unconditionally.

The conditions for sending the notification have changed slightly since the revision you placed this comment on, but you are right that the path write could be conditional. I think I just didn't want to add an extra branch; functionally it doesn't matter.